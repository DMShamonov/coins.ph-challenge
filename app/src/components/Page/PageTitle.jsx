// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PageTitlePropsType } from './types.js.flow';

export default function PageTitle({
  className,
  children,
}: PageTitlePropsType): React.Node {
  return (
    <h1 className={classnames('page__title', className)}>
      {children}
    </h1>
  );
}
