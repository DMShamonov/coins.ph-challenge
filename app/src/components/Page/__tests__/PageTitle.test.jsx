import React from 'react';
import { shallow } from 'enzyme';

import PageTitle from '../PageTitle';

describe('`PageTitle` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<PageTitle>__CONTENT__</PageTitle>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<PageTitle className="__CLASSNAME__">__CONTENT__</PageTitle>)).toMatchSnapshot();
  });
});
