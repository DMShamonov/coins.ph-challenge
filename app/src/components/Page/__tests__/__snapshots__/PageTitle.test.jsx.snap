// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`PageTitle\` component should render correctly with \`className\` prop 1`] = `
<h1
  className="page__title __CLASSNAME__"
>
  __CONTENT__
</h1>
`;

exports[`\`PageTitle\` component should render correctly with default props 1`] = `
<h1
  className="page__title"
>
  __CONTENT__
</h1>
`;
